wire-ios-sync-engine/Source/Use cases/GetSelfUserVerificationStatusUseCase.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+// sourcery: AutoMockable
+public protocol GetSelfUserVerificationStatusUseCaseProtocol {
+    func invoke() async throws -> (isMLSCertified: Bool, isProteusVerified: Bool)
+}
+
+public struct GetSelfUserVerificationStatusUseCase: GetSelfUserVerificationStatusUseCaseProtocol {
+
+    private let context: NSManagedObjectContext
+    private let coreCryptoProvider: CoreCryptoProviderProtocol
+
+    public init(
+        context: NSManagedObjectContext,
+        coreCryptoProvider: CoreCryptoProviderProtocol
+    ) {
+        self.context = context
```

Comment:
**(blocking)**
Q: Would this run on any context?  
**suggestion:** If this needs to be run on `sync context` .May be have a precondition to check the context.

Reply:
Why should we have this restriction? It would be nice if the use case works for any context.

Reply:
I added `context.perform { ... }`, so the use cases should work for any context.